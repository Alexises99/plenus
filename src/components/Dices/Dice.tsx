import { ReactNode } from 'react'

interface DiceProps {
  children: ReactNode
  active?: boolean
}

export default function Dice({ active = false, children }: DiceProps) {
  if (active) {
    return (
      <div className="border-2 rounded-full aspect-square h-8 lg:h-20 flex justify-center items-center cursor-pointer">
        <span className="text-xl lg:text-3xl font-bold text-white flex justify-center items-center w-full">
          {children}
        </span>
      </div>
    )
  } else {
    return (
      <div className="border-2 rounded-full aspect-square h-12 lg:h-20 flex justify-center items-center cursor-pointer">
        <span className="text-xl lg:text-3xl font-bold text-white flex justify-center items-center w-full">
          {children}
        </span>
      </div>
    )
  }
}
