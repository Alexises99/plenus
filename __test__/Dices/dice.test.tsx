import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Dice from '@/components/Dices/Dice'

// const dice: DiceData = {
//   diceColor: 'red',
//   diceNumber: 2
// }

const testNumber = 1

describe('Dice', () => {
  it('renders something', () => {
    render(<Dice active>{testNumber}</Dice>)

    const number = screen.getByText(testNumber)

    expect(number).toBeInTheDocument()
  })

  it('renders dice unchanged', () => {
    const { container } = render(<Dice active>{testNumber}</Dice>)
    expect(container).toMatchSnapshot()
  })
})
