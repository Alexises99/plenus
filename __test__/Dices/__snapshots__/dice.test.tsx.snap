// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dice renders dice unchanged 1`] = `
<div>
  <div
    class="border-2 rounded-full aspect-square h-8 lg:h-20 flex justify-center items-center cursor-pointer"
  >
    <span
      class="text-xl lg:text-3xl font-bold text-white flex justify-center items-center w-full"
    >
      1
    </span>
  </div>
</div>
`;
